cell_classification/evaluation_script.py

```diff
@@ -17,51 +15,56 @@
         "--model_path",
         type=str,
         help="Path to model weights",
-        default="C:\\Users\\lorenz\\Desktop\\angelo_lab\\cell_classification\\checkpoints\\" +
-        "ex_2\\ex_2.h5",
+        default=None,
     )
     parser.add_argument(
         "--params_path",
         type=str,
         help="Path to model params",
-        default="C:\\Users\\lorenz\\Desktop\\angelo_lab\\cell_classification\\checkpoints\\"
-        + "ex_2\\params.toml",
+        default="E:\\angelo_lab\\test\\params.toml",
+    )
+    parser.add_argument(
+        "--worst_n",
+        type=int,
+        help="Number of worst predictions to plot",
+        default=20,
     )
     args = parser.parse_args()
     with open(args.params_path, "r") as f:
         params = toml.load(f)
-    params["model_path"] = args.model_path
+    if args.model_path is not None:
+        params["model_path"] = args.model_path
     model, val_dset = load_model_and_val_data(params)
     params["eval_dir"] = os.path.join(*os.path.split(params["model_path"])[:-1], "eval")
     os.makedirs(params["eval_dir"], exist_ok=True)
     pred_list = model.predict_dataset(val_dset, False)
 
-    # pixel level evaluation
-    print("Calculate ROC curve")
-    roc = calc_roc(pred_list)
-    with open(os.path.join(params["eval_dir"], "roc.pkl"), "wb") as f:
-        pickle.dump(roc, f)
-    tprs, mean_tprs, fpr, std, mean_thresh = average_roc(roc)
-    plot_average_roc(mean_tprs, std, save_dir=params["eval_dir"], save_file="avg_roc.png")
-    print("AUC: {}".format(np.mean(roc["auc"])))
-
-    # print("Calculate precision, recall, f1_score and accuracy")
-    # avg_metrics = calc_metrics(pred_list)
-    # pd.DataFrame(avg_metrics).to_csv(
-    #     os.path.join(params["eval_dir"], "pixel_metrics.csv"), index=False
-    # )
-    # plot_metrics_against_threshold(
-    #     avg_metrics,
-    #     metric_keys=["precision", "recall", "f1_score"],
-    #     threshold_key="threshold",
-    #     save_dir=params["eval_dir"],
-    #     save_file="precision_recall_f1.png",
-    # )
+    # prepare cell_table
+    activity_list = []
+    for pred in pred_list:
+        activity_df = pred["activity_df"].copy()
+        for key in ["dataset", "marker", "folder_name"]:
+            activity_df[key] = [pred[key]]*len(activity_df)
+        activity_list.append(activity_df)
+    activity_df = pd.concat(activity_list)
+    activity_df.to_csv(os.path.join(params["eval_dir"], "pred_cell_table.csv"), index=False)
 
     # cell level evaluation
     roc = calc_roc(pred_list, gt_key="activity", pred_key="pred_activity", cell_level=True)
     with open(os.path.join(params["eval_dir"], "roc_cell_lvl.pkl"), "wb") as f:
         pickle.dump(roc, f)
+
+    # find index of n worst predictions and save plots of them
+    worst_idx = np.argsort(roc["auc"])[-args.worst_n:]
```

Comment:
I think there should be the option to do this by marker, in addition to global `worst_n`